.github/workflows/release-build-publish.yml

```diff
@@ -50,95 +50,95 @@ jobs:
-#        uses: docker/build-push-action@v2
-#        if: ${{ steps.prep.outputs.branch == github.event.repository.default_branch }}
-#        with:
-#          file: docker/Dockerfile-database
-#          context: .
-#          push: true
-#          tags: provenanceio/digital-currency-consortium-database:latest
+  docker_build:
+    name: Build and Push to Docker Registry
+    needs: java_build
+
+    runs-on: ubuntu-latest
+
+    if: github.event_name == 'release'
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Prepare
+        id: prep
+        run: |
+          # Strip git ref prefix from version
+          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+
+          # Strip "v" prefix from tag name
+          [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
+
+          # Use Docker `latest` tag convention
+          [ "$VERSION" == "main" ] && VERSION=latest
+
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+
+          SERVICE_IMAGE_ID=figuretechnologies/digital-currency-consortium-service
+          REPORT_IMAGE_ID=figuretechnologies/digital-currency-consortium-report
+
+          SERVICE_TAGS=$SERVICE_IMAGE_ID:$VERSION
+          REPORT_TAGS=$REPORT_IMAGE_ID:$VERSION
+
+          PRERELEASE=${{ github.event.release.prerelease }}
+          echo PRERELEASE=$PRERELEASE
+
+          if [ "$PRERELEASE" == "false" ]; then
+              SERVICE_TAGS=$SERVICE_TAGS,$SERVICE_IMAGE_ID:latest
+              REPORT_TAGS=$REPORT_TAGS,$REPORT_IMAGE_ID:latest
+          fi
+
+          echo ::set-output name=version::${VERSION}
+          echo ::set-output name=service_tags::${SERVICE_TAGS}
+          echo ::set-output name=report_tags::${REPORT_TAGS}
+          echo ::set-output name=created::$(date -u +'%Y-%m-%dT%H:%M:%SZ')
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+        with:
+          install: true
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Download Service Artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: service
+          path: service/build/libs
+
+      - name: Build and push service docker
+        uses: docker/build-push-action@v3
+        with:
+          file: docker/Dockerfile-service
+          context: .
+          push: true
+          tags: ${{ steps.prep.outputs.service_tags }}
+
+      - name: Download Report Artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: report
+          path: report/build/libs
+
+      - name: Build and push report docker
+        uses: docker/build-push-action@v3
+        with:
+          file: docker/Dockerfile-report
+          context: .
+          push: true
+          tags: ${{ steps.prep.outputs.report_tags }}
+
```

Comment:
do we not need a log into dockerhub step from secrets or do we not need that? thought i saw that on object store gateway

Reply:
doh nvm its above :disappear: